dev/modules/handle-click.js

```diff
@@ -0,0 +1,124 @@
+import { colorLuminance } from './utils';
+import { getModal } from './handle-swal-dom';
+import { hasClass, isDescendant } from './handle-dom';
+
+
+/*
+ * User clicked on "Confirm"/"OK" or "Cancel"
+ */
+var handleButton = function(event, params, modal) {
+  var e = event || window.event;
+  var target = e.target || e.srcElement;
+
+  var targetedConfirm = target.className.indexOf('confirm') !== -1;
+  var targetedOverlay = target.className.indexOf('sweet-overlay') !== -1;
+  var modalIsVisible  = hasClass(modal, 'visible');
+  var doneFunctionExists = (params.doneFunction && modal.getAttribute('data-has-done-function') === 'true');
+
+  // Since the user can change the background-color of the confirm button programmatically,
+  // we must calculate what the color should be on hover/active
+  var normalColor, hoverColor, activeColor;
+  if (targetedConfirm && params.confirmButtonColor) {
+    normalColor  = params.confirmButtonColor;
+    hoverColor   = colorLuminance(normalColor, -0.04);
+    activeColor  = colorLuminance(normalColor, -0.14);
+  }
+
+  function shouldSetConfirmButtonColor(color) {
+    if (targetedConfirm && params.confirmButtonColor) {
+      target.style.backgroundColor = color;
+    }
+  }
+
+  switch (e.type) {
+    case 'mouseover':
+      shouldSetConfirmButtonColor(hoverColor);
+      break;
+
+    case 'mouseout':
+      shouldSetConfirmButtonColor(normalColor);
+      break;
+
+    case 'mousedown':
+      shouldSetConfirmButtonColor(activeColor);
+      break;
+
+    case 'mouseup':
+      shouldSetConfirmButtonColor(hoverColor);
+      break;
+
+    case 'focus':
+      let $confirmButton = modal.querySelector('button.confirm');
+      let $cancelButton  = modal.querySelector('button.cancel');
+
+      if (targetedConfirm) {
+        $cancelButton.style.boxShadow = 'none';
+      } else {
+        $confirmButton.style.boxShadow = 'none';
+      }
+      break;
+
+    case 'click':
+      let clickedOnModal = (modal === target);
```

Comment:
'let' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).